lib/rex/post/meterpreter/extensions/stdapi/railgun/def/windows/def_wldap32.rb

```diff
@@ -26,13 +26,13 @@ def self.create_library(constant_manager, library_path = 'wldap32')
     ], 'ldap_bind_sA', "cdecl")
 
     dll.add_function('ldap_search_sA', 'DWORD',[
-        ['DWORD', 'ld', 'in'],
+        ['PHANDLE', 'ld', 'in'],
         ['PCHAR', 'base', 'in'],
         ['DWORD', 'scope', 'in'],
         ['PCHAR', 'filter', 'in'],
         ['PCHAR', 'attrs[]', 'in'],
         ['DWORD', 'attrsonly', 'in'],
```

Comment:
It would also make sense to add an alias for DWORD to ULONG for scope and attrs only here so the definition would be a [closer match.](https://learn.microsoft.com/en-us/windows/win32/api/winldap/nf-winldap-ldap_search_sa).

The `PZPSTR` is the problem here since we don't have support for
> A null-terminated array of null-terminated strings indicating the attributes to return for each matching entry. Pass NULL to retrieve all available attributes.

Adding that wouldn't be too bad but let's leave that out for now. We can revisit it if we see that `PZPSTR` is used in more places.

Reply:
Attempted to fix this with https://github.com/rapid7/metasploit-framework/pull/17562/commits/18bad9d75e770b405686397af7c1f6c9ae6cde87. Let me know if that works for you.